pkg/upgrade/templates/common.go

```diff
@@ -0,0 +1,28 @@
+/*
```

Comment:
Just thinking aloud on how this can be extended in the future upgrades. Consider the following case. 
- Upgrading from 1.2 to 1.3 -- requires some additional changes to the patch template of CSP or jiva. 
- Some customers would like to upgrade to 1.0 to 1.2 only - and those upgrades need to apply the old patch template. So we will need to have a mechanism to invoke templates based on the version to which an object is being upgraded to. 

Probably one way to solve this is to version the templates. For example move this package to `pkg/upgrade/templates/v1` 

v1 hosts the current templates. 
v2 can host the modified templates in future releases -- say from 1.3 to 1.5
v3 can host further templates 

Whenever there is a change in the templates a new version folder can be created. 

Depending on the version to upgrade, the right template version can be picked. For the above example, this would look like:
if upgrading to: 1.1 or 1.2 -> v1 template
if upgrading to: 1.3 or 1.4 or 1.5 -> v2 template

Reply:
how will we manage if there are templates common between v1 and v2 and some are common between v2 and v3 and nothing is common between v1 and v3. 
 - we should maintain all three folders with no repetition of templates.
**or**
 - we should have the repetition and maintain only current folder we are working with.

If we go with second approach we can clean the code afterwards easily when we stop support for any upgrade.

Reply:
Done

Reply:
Let us evaluate if we should use interface to manage this. v1, v2 folders can be optional.
For example:

```go
// pkg/upgrade/interface.go
type Interface interface {
  Upgrade() error
}
```
```go
// pkg/upgrade/jiva.go

type JivaPatchTpl string

// used for versions 1.0 & 1.1
const JivaPatchV1 JivaPatchTpl = `...`

type JivaPatch struct {
  values map[string]string
  template JivaPatchV1
}

func NewJivaPatch(tpl JivaPatchTpl, values map[string]string) *JivaPatch {
  return &JivaPatch{
    values: values,
    template: tpl,
  }
}

func (j *jivapatch) Upgrade() error {
 // logic
}
```

Reply:
The approach looks good. Will take this in a separate PR.